crates/compiler/mono/src/ir/pattern.rs

```diff
@@ -1526,6 +1544,54 @@ fn store_list_pattern<'a>(
         }
     }
 
+    if let Some((index, Some(rest_sym))) = opt_rest {
```

Comment:
I really want us to modularize mono more. Definitely a nit for this feature but could you put this in a separate function?

Reply:
Extracted